### PR TITLE
API Gateway and gRPC client

### PR DESCRIPTION
**Pull request by bgonilho**

### CODE REVIEW

File: services/gateway/gRPC_client.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import grpc
+import user_pb2
+import user_pb2_grpc
+
+class GRPCClient(object):
+    def __init__(self):
+        self.host = '127.0.0.1'
```
Comment (luisferreira32) [inline services/gateway/gRPC_client.py]: Make these (address and port) come from env at least

File: services/gateway/gRPC_client.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        self.server_port = '8000'
+
+        self.channel = grpc.insecure_channel('{}:{}'.format(self.host, self.server_port))
```
Comment (luisferreira32) [inline services/gateway/gRPC_client.py]: Would these be equivalent? We should define one form of formatting strings and stick with it across the code base.
```suggestion
        self.channel = grpc.insecure_channel(f'{self.host}:{self.server_port}')
```

File: services/gateway/gRPC_client.py
Comment (luisferreira32) [inline services/gateway/gRPC_client.py:1]: Ah, this is a test file, maybe it does not make sense to be commited in the service, keep it to yourself or port it to some "integration tests draft" already in `test/integration-test`

File: services/gateway/server.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+async def root():
+    return {"message":"Hello World!"}
+
+def authenticate_user(username, hashed_password):
+    with grpc.insecure_channel("localhost:" + config("MS2_PORT")) as channel:
+        stub = user_pb2_grpc.UserServiceStub(channel = channel)
+        response = stub.Login(
+            user_pb2.LoginRequest(username = username, hashed_password = hashed_password)
+        )
+        return response.auth_token
```
Comment (luisferreira32) [inline services/gateway/server.py]: The front-end should be looking for more than an auth token, missing refresh and ID.

File: services/gateway/server.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+app = FastAPI()
+
+@app.get("/")
+async def root():
+    return {"message":"Hello World!"}
+
+def authenticate_user(username, hashed_password):
+    with grpc.insecure_channel("localhost:" + config("MS2_PORT")) as channel:
+        stub = user_pb2_grpc.UserServiceStub(channel = channel)
+        response = stub.Login(
```
Comment (luisferreira32) [inline services/gateway/server.py]: You should read the context of the answer from the response, iirc, there ought to be something where you'll get some errors and might want to propagate them as an expection? Not really sure how python gRPC would work here, we have to try it out.

File: services/gateway/testclient.py
Comment (luisferreira32) [inline services/gateway/testclient.py:1]: Another test client that should not be in the service

File: services/user/justfile [hunk 5]
```diff
@@ -1,4 +1,6 @@
 set dotenv-load := false
+venv:
+    if not exist venv ( python -m venv venv && .\venv\Scripts\activate && pip install -r requirements.dev.txt )
```
Comment (luisferreira32) [inline services/user/justfile:3]: This is highly based on cmd usage... remove! 😠

**Review by luisferreira32 — CHANGES_REQUESTED** [review]
Nice first PR! Some minor comments, feel free to question them as well :+1:

File: services/gateway/gRPC_client.py (hunk 1 above)
Comment (LionOfGraphs) [reply services/gateway/gRPC_client.py]: Please check other services and make use of the _python-decouple_ library.

File: services/gateway/gRPC_client.py (hunk 2 above)
Comment (luisferreira32) [reply services/gateway/gRPC_client.py]: Honest question here.

Comment (bgonilho) [reply services/gateway/gRPC_client.py]: This file will not be used

File: services/user/justfile (hunk 5 above)
Comment (bgonilho) [reply services/user/justfile:3]: Generalization of just file can be done in a future PR

**Review by antoniomf97 — APPROVED** [review]
Good good

**Review by luisferreira32 — APPROVED** [review]
blind approve 👀

**Pull request merged**